tests/migrations/test_update_mlwh_with_legacy_samples.py

```diff
@@ -0,0 +1,162 @@
+import sys
+import traceback
+from migrations.helpers.mlwh_samples_update_helper import (
+    valid_datetime_string,
+    update_mlwh_with_legacy_samples,
+)
+from crawler.constants import (
+    FIELD_ROOT_SAMPLE_ID,
+    FIELD_RNA_ID,
+    FIELD_RESULT,
+    FIELD_DATE_TESTED,
+    FIELD_LAB_ID,
+    FIELD_SOURCE,
+    FIELD_PLATE_BARCODE,
+    FIELD_CREATED_AT,
+    FIELD_UPDATED_AT,
+    MLWH_TABLE_NAME,
+    MONGO_DATETIME_FORMAT,
+)
+import pytest
+from datetime import (datetime, timedelta)
+
+def print_exception(e: Exception) -> None:
+    print(f'An exception occurred, at {datetime.now()}')
+    e = sys.exc_info()
+    print(e[0]) # exception type
+    print(e[1]) # exception message
+    if e[2]: # traceback
+      traceback.print_tb(e[2], limit=10)
+
+def generate_example_samples(range, start_datetime):
+    samples = []
+    for n in range:
+        samples.append({
+            FIELD_ROOT_SAMPLE_ID: "TLS0000000" + str(n),
+            FIELD_RNA_ID: "TL-rna-00000001_A01",
+            FIELD_RESULT: "Positive",
+            FIELD_PLATE_BARCODE: "DN1000000" + str(n),
+            FIELD_DATE_TESTED: "2020-05-10 14:01:00 UTC",
+            FIELD_LAB_ID: "TLS",
+            FIELD_SOURCE: "Test Lab Somewhere",
+            FIELD_CREATED_AT: start_datetime + timedelta(days=n),
+            FIELD_UPDATED_AT: start_datetime + timedelta(days=n),
+        })
+
+    print(samples)
+    return samples
+
+def clear_mlwh_table(mysql_conn):
+    print("Clearing MLWH database table")
+
+    cursor = mysql_conn.cursor()
+    try:
+        cursor.execute(f"TRUNCATE TABLE unified_warehouse_test.{MLWH_TABLE_NAME}")
```

Comment:
Should use MLWH_DB_DBNAME, since it's in the constants?

Reply:
Truncate table wipes the table of all data. I thought that would be very bad if it was run on prod, so hardcoded the test database.. What do you think though? any way a test can be run against anything other than a test database?

Reply:
Changed